src/app/leverage/page.tsx

```diff
@@ -1,27 +1,29 @@
-'use client'
-
-import { useEffect } from 'react'
+import { headers } from 'next/headers'
 
+import { LeverageOnboardingPopupWrapper } from '@/app/leverage/components/leverage-onboarding-popup'
 import { LeveragePanel } from '@/app/leverage/components/leverage-panel'
+import { getProtection } from '@/lib/utils/protection'
 
+import { BodyClassEffect } from './components/body-class-effect'
 import { FaqSection } from './components/faq-section'
 
-export default function Page() {
-  useEffect(() => {
-    document.body.classList.add('dark', 'bg-zinc-950')
-    return () => {
-      document.body.classList.remove('dark', 'bg-zinc-950')
-    }
-  }, [])
+export default async function Page() {
+  const hdrs = headers()
+  const headersObj = new Headers(hdrs)
+  const { isNewUser } = await getProtection(headersObj)
```

Comment:
Made the page a server component to be able to pre-fetch the user state, so that we can determine right away if the leverage onboarding popup should be shown or not.